### PR TITLE
Add PureConfig support

### PR DESCRIPTION
**Pull request by jcazevedo**
This PR adds [PureConfig](https://github.com/melrief/pureconfig) support to `ApsoConfig`, so that it now supports the `to[T]` method for types which have the `ConfigConvert` typeclass.

We might consider deprecating the methods that use the `ConfigReader` typeclass in the future.

### CODE REVIEW

File: apso/src/main/scala/eu/shiftforward/apso/config/Implicits.scala [hunk 1]
```diff
@@ -328,8 +330,10 @@ object Implicits extends BasicConfigReaders with ExtendedConfigReaders {
         value <- Try(conf.getConfig(key)).toOption
       } yield { key -> value }).toMap
     }
-  }
 
+    def to[T: ConfigConvert]: Try[T] =
+      loadConfig[T](pcConf.typesafeConfigToConfig(conf))
```
Comment (ruippeixotog) [inline apso/src/main/scala/eu/shiftforward/apso/config/Implicits.scala]: You can do just `loadConfig(conf)` here.

Comment (jcazevedo) [inline apso/src/main/scala/eu/shiftforward/apso/config/Implicits.scala]: Yeah. This was only recently fixed in PureConfig. I'll update it.

Comment (jcazevedo) [inline apso/src/main/scala/eu/shiftforward/apso/config/Implicits.scala]: Fixed in 53c7ff9.

**Comment by ruippeixotog** [comment]
Why don't we deprecate methods using `ConfigReader` now? That would encourage people to update to PureConfig based methods gradually.

**Comment by jcazevedo** [comment]
Let's leave the deprecation for later, when we've fully adopted `PureConfig`.

**Comment by ruippeixotog** [comment]
Roger! LGTM then

**Pull request merged**